src/contrib/cluster/Akka.Cluster.Sharding/PersistentShardCoordinator.cs

```diff
@@ -109,12 +109,15 @@ public IImmutableSet<ShardId> AllShards
             }
 
             /// <summary>
-            /// TBD
+            /// Feed an event into the ShardCoordinator state.
             /// </summary>
-            /// <param name="e">TBD</param>
-            /// <exception cref="ArgumentException">TBD</exception>
-            /// <returns>TBD</returns>
-            public State Updated(IDomainEvent e)
+            /// <param name="e">The event to process.</param>
+            /// <param name="isRecovering">A flag to indicate if we're trying to recover state previously stored in the database.
+            /// We need to be more tolerant when this happens in the name of trying to accelerate recovery, so the system doesn't compromise
+            /// itself and go offline.</param>
+            /// <exception cref="ArgumentException">Thrown if an event is illegal in the current state.</exception>
+            /// <returns>An update copy of this state.</returns>
+            public State Updated(IDomainEvent e, bool isRecovering = false)
```

Comment:
We now accept a new parameter here, `isRecovering` - when this value is set to `true` then we enable "reader tolerance" inside the `PersistentShardCoordinator`.